torch/utils/benchmark/utils/cpp_jit.py

```diff
@@ -66,3 +89,55 @@ def get_compat_bindings() -> CallgrindModuleType:
                 extra_include_paths=EXTRA_INCLUDE_PATHS,
             )
     return COMPAT_CALLGRIND_BINDINGS
+
+
+def _compile_template(stmt: str, setup: str, src: str, is_standalone: bool) -> Any:
+    for before, after, indentation in (
+        ("// SETUP_TEMPLATE_LOCATION", setup, 4),
+        ("// STMT_TEMPLATE_LOCATION", stmt, 8)
+    ):
+        # C++ doesn't care about indentation so this code isn't load
+        # bearing the way it is with Python, but this makes the source
+        # look nicer if a human has to look at it.
+        src = re.sub(
+            before,
+            textwrap.indent(after, " " * indentation)[indentation:],
+            src
+        )
+
+    # We want to isolate different Timers. However `cpp_extension` will
+    # cache builds which will significantly reduce the cost of repeated
+    # invocations.
+    with LOCK:
+        if not os.path.exists(BUILD_ROOT):
+            os.makedirs(BUILD_ROOT)
+            atexit.register(shutil.rmtree, BUILD_ROOT)
```

Comment:
https://github.com/pytorch/pytorch/pull/47428/files#r521094904

Reply:
I'm implicitly relying on `torch._appdirs.user_cache_dir` having similar kinds of litter safety. (That's what you would get from vanilla `cpp_extension` anyway) On my machine it's `~/.cache/...`, for instance. And if you're still unconvinced it's no trouble to change it to use `tempfile` instead.

Reply:
https://askubuntu.com/questions/102046/is-it-okay-to-delete-the-cache-folder seems to suggest that `~/.cache` doesn't get automatically cleaned.